File: file_manipulator.py
import sys


class FileManipulator:
    def __init__(self, input_file=None, output_file=None):
        self.input_file = input_file
        self.output_file = output_file

    def reverse(self, input_file, output_file):
        with open(input_file, 'r') as in_file:
            original_text = in_file.read()

        with open(output_file, 'w') as out_file:
            reversed_text = ''.join(reversed(original_text))
            out_file.write(reversed_text)

    def copy(self, input_file, output_file):
        with open(input_file, 'r') as in_file:
            original_text = in_file.read()

        with open(output_file, 'w') as out_file:
            out_file.write(original_text)

    def duplicate(self, input_file, n):
        with open(input_file, 'r+') as in_file:
            original_text = in_file.read()
            duplicated_text = original_text * int(n)

        with open(input_file, 'w') as in_file:
            in_file.write(duplicated_text)

    def replace_string(self, input_file, needle, new_string):
        with open(input_file, 'r') as in_file:
            original_text = in_file.read()
            replaced_string = original_text.replace(needle, new_string)

        with open(input_file, 'w') as in_file:
            in_file.write(replaced_string)


def main():
    if len(sys.argv) < 4:
        print("Usage: program <command> <inputFile> <arguments...>")
        sys.exit(1)

    command = sys.argv[1]
    args = sys.argv[2:]

    file_manipulator = FileManipulator()

    if command == 'reverse':
        file_manipulator.reverse(*args)
    elif command == 'copy':
        file_manipulator.copy(*args)
    elif command == 'duplicate':
        file_manipulator.duplicate(*args)
    elif command == 'replace-string':
        file_manipulator.replace_string(*args)
    else:
        print("Invalid command")
        sys.exit(1)

    print("Operation completed successfully.")

if __name__ == '__main__':
    main()

File: file-manipulator.go
package main

import (
	"fmt"
	"io"
	"os"
	"strconv"
	"strings"
)

type FileManipulator struct {
}

func handleError(err error) {
	if err != nil {
		fmt.Println("Error:", err)
		os.Exit(1)
	}
}

func (f FileManipulator) reveres(inputPath, outputPath string) error {
	inFile, err := os.Open(inputPath)
	handleError(err)
	defer inFile.Close()

	outFile, err := os.Create(outputPath)
	handleError(err)
	defer outFile.Close()

	buf := make([]byte, 1024)
	for {
		n, err := inFile.Read(buf)
		if err != nil && err != io.EOF {
			return err
		}
		if n == 0 {
			break
		}
		for i := n - 1; i >= 0; i-- {
			_, err := outFile.Write([]byte{buf[i]})
			if err != nil {
				return err
			}
		}
	}

	return nil
}

func (f FileManipulator) copy(inputPath, outputPath string) error {
	inFile, err := os.Open(inputPath)
	handleError(err)
	defer inFile.Close()

	outFile, err := os.Create(outputPath)
	handleError(err)
	defer outFile.Close()

	_, err = io.Copy(outFile, inFile)
	handleError(err)

	return nil
}

func (f FileManipulator) duplicate(inputPath, n string) error {
	content, err := os.ReadFile(inputPath)
	handleError(err)

	var newContent []byte
	nInt, _ := strconv.Atoi(n)

	for i := 0; i < nInt; i++ {
		newContent = append(newContent, content...)
	}

	err = os.WriteFile(inputPath, newContent, 0644)
	handleError(err)

	return nil
}

func (f FileManipulator) replaceString(inputPath, needle, newString string) error {
	content, err := os.ReadFile(inputPath)
	handleError(err)

	str := string(content)
	replaced := strings.Replace(str, needle, newString, -1)
	bytesStr := []byte(replaced)

	err = os.WriteFile(inputPath, bytesStr, 0644)
	handleError(err)

	return nil
}

func main() {
	if len(os.Args) < 4 {
		fmt.Println("Usage: program <command> <inputFile> <arguments...>")
		os.Exit(1)
	}

	command := os.Args[1]
	args := os.Args[2:]

	f := FileManipulator{}

	switch command {
	case "reverse":
		handleError(f.reveres(args[0], args[1]))
	case "copy":
		handleError(f.copy(args[0], args[1]))
	case "duplicate":
		handleError(f.duplicate(args[0], args[1]))
	case "replace-string":
		handleError(f.replaceString(args[0], args[1], args[2]))
	default:
		fmt.Println("Invalid command.")
		os.Exit(1)
	}

	fmt.Println("Operation completed successfully.")
}
